### PR TITLE
BCP-006-01: Test suites for NMOS With JPEG XS

### PR DESCRIPTION
**Pull request by garethsb**
Addresses #714.

Replaces #715 with a **bcp-006-01** activity branch in the AMWA-TV/nmos-testing for simpler collaboration. @jonathan-r-thorpe, @joostrovers, @rbgodwin-nt, @David-P-B, etc. can make PRs to extend the test suites onto this branch, which the activity group can review. @alabou, @cristian-recoseanu, etc. can use this 'stable' activity branch for testing.

### CODE REVIEW

**Comment by cristian-recoseanu** [comment]
Re-tested the controller test suite today. Working fine but I did have some comments.

Does it make sense to have the `MAX_TEST_ITERATIONS` configuration item by default to something which isn't 0? This would mean at JT-NM if someone is using this suite in anger they don't get frustrated with having to go through every single test scenario because they don't know about the configuration item.

I've ran the test suite a couple of times and each time test_01 and test_02 always had all the options presented as jxsv. I guess this is because we register so many more jsxv resources compared to video/raw that the probability is skewed towards jxsv. Does it make sense for test_01 and test_02 to always sneak in a video/raw option in there amongst all the other jxsv options randomly selected?

**Comment by garethsb** [comment]
> Does it make sense to have the MAX_TEST_ITERATIONS configuration item by default to something which isn't 0? This would mean at JT-NM if someone is using this suite in anger they don't get frustrated with having to go through every single test scenario because they don't know about the configuration item.

My thinking is to keep the default as comprehensive, but to highlight better in the doc and while running the tool that it can be changed. Otherwise people definitely won't test comprehensively and when it comes to JT-NM Tested they will say "it passed everything when I ran it".

File: nmostesting/NMOSTesting.py [hunk 1]
```diff
@@ -561,7 +598,7 @@ def run_tests(test, endpoints, test_selection=["all"]):
         elif test == "IS-09-02":
             # This test has an unusual constructor as it requires a system api instance
             test_obj = test_def["class"](apis, SYSTEMS, DNS_SERVER)
-        elif test == "IS-04-04" or test == "IS-05-03":
+        elif test in ["IS-04-04", "IS-05-03", "BCP-006-01-02"]:
```
Comment (garethsb) [inline nmostesting/NMOSTesting.py:601]: @jonathan-r-thorpe if all/most Controller test suites will follow the same form, would it make sense to use `isinstance(test_def["class"], ControllerTest)` here?

**Comment by garethsb** [comment]
> I've ran the test suite a couple of times and each time test_01 and test_02 always had all the options presented as jxsv.

Hmmm, that looks like the way it's set up at the moment. The candidate senders are always fully populated with JPEG XS ones if possible (and it's always possible). @jonathan-r-thorpe, I assume this isn't what you actually intended?

File: nmostesting/NMOSTesting.py (hunk 1 above)
Comment (jonathan-r-thorpe) [reply nmostesting/NMOSTesting.py:601]: Yes, that is a much better/more maintainable approach.

**Comment by jonathan-r-thorpe** [comment]
Yes, that's a bug that's crept in - it should have at least 1 JXSV and 1 RAW in the presented options - I will check and fix

**Comment by jonathan-r-thorpe** [comment]
The lack of "video/raw" example Senders/Receivers has been addressed in #740

**Comment by garethsb** [comment]
Looking at the test plan Google sheet, I think we've now covered everything except the **IS-04 Receivers** tests and the **IS-05** tests?

**Comment by garethsb** [comment]
Is IS-05-02 test_13 already a sufficient test of the IS-05 Sender _/transportfile_ endpoint for BCP-006-01? It checks that file contents is identical to the IS-04 `manifest_href`, which we are already testing thoroughly?

https://github.com/AMWA-TV/nmos-testing/blob/cc71d2be1245a28153297df3bb1b141e24863405/nmostesting/suites/IS0502Test.py#L618

**Comment by garethsb** [comment]
Any Receiver `caps.constraint_sets` parameter constraints' correct target types are already checked by IS-04-01 test_27_1.

https://github.com/AMWA-TV/nmos-testing/blob/cc71d2be1245a28153297df3bb1b141e24863405/nmostesting/suites/IS0401Test.py#L1720

So maybe the only checks required in the BCP-006-01 suite could be that the string ones use valid `enum` values...

**Review by alabou — DISMISSED** [review]

**Review by peterbrightwell — APPROVED** [review]
Looks good, to the extent that I across it but there are parts that I don't feel qualify to comment on.

**Pull request merged**